README
NAME
    Dist::Zilla::PluginBundle::MITHALDU - Dist::Zilla configuration the way
    MITHALDU does it

VERSION
    version 1.141830

SYNOPSIS
       # in dist.ini
       [@MITHALDU]

DESCRIPTION
    This module is forked off of Dist::Zilla::PluginBundle::DAGOLDEN and
    modified to suit my own tastes. He did most of the work.

    This is a Dist::Zilla PluginBundle. It is roughly equivalent to the
    following dist.ini:

       ; version provider
       [AutoVersion]  ; build a version from the date
       major = 1
 
       ; choose files to include
       [GatherDir]         ; everything under top dir
       exclude_filename = README.pod   ; skip this generated file
       exclude_filename = META.json    ; skip this generated file
       exclude_filename = .gitignore   ; skip this generated file
       exclude_filename = README.PATCHING ; skip this generated file
       exclude_filename = perlcritic.rc   ; skip this generated file
 
       [PruneCruft]        ; default stuff to skip
       except = .gitignore
       except = README.PATCHING
       except = perlcritic.rc
       [ManifestSkip]      ; if -f MANIFEST.SKIP, skip those, too
 
       ; file modifications
       [OurPkgVersion]     ; add $VERSION = ... to all files
       [InsertCopyright    ; add copyright at "# COPYRIGHT"
       [PodWeaver]         ; generate Pod
       config_plugin = @MITHALDU ; my own plugin allows Pod::WikiDoc
 
       ; generated files
       [License]           ; boilerplate license
       [ReadmeFromPod]     ; from Pod (runs after PodWeaver)
       [ReadmeAnyFromPod]  ; create README.pod in repo directory
       type = pod
       filename = README.pod
       location = root
       [GenerateFile]
       filename    = .gitignore
       is_template = 1
       content = /.build
       content = /{{ $dist->name }}-*
       ; and more, see Dist::Zilla::PluginBundle::MITHALDU::Templates
       [GenerateFile]
       filename    = README.PATCHING
       is_template = 1
       content = README.PATCHING
       ; and more, see Dist::Zilla::PluginBundle::MITHALDU::Templates
       [GenerateFile]
       filename    = perlcritic.rc
       is_template = 1
       content = README.PATCHING
       ; and more, see Dist::Zilla::PluginBundle::MITHALDU::Templates
 
       ; t tests
       [Test::Compile]     ; make sure .pm files all compile
       fake_home = 1       ; fakes $ENV{HOME} just in case
 
       ; xt tests
       [Test::PodSpelling] ; xt/author/pod-spell.t
       [Test::Perl::Critic]; xt/author/critic.t
       [MetaTests]         ; xt/release/meta-yaml.t
       [PodSyntaxTests]    ; xt/release/pod-syntax.t
       [PodCoverageTests]  ; xt/release/pod-coverage.t
       [Test::Portability] ; xt/release/portability.t (of file name)
       [Test::Version]     ; xt/release/test-version.t
 
       ; metadata
       [AutoPrereqs]       ; find prereqs from code
       [MinimumPerl]       ; determine minimum perl version
       [GithubMeta]
 
       [MetaNoIndex]       ; sets 'no_index' in META
       directory = t
       directory = xt
       directory = examples
       directory = corpus
       package = DB        ; just in case
 
       [Bugtracker]        ; defaults to RT
 
       [MetaProvides::Package] ; add 'provides' to META files
       meta_noindex = 1        ; respect prior no_index directives
 
       [MetaYAML]          ; generate META.yml (v1.4)
       [MetaJSON]          ; generate META.json (v2)
 
       ; build system
       [ExecDir]           ; include 'bin/*' as executables
       [ShareDir]          ; include 'share/' for File::ShareDir
       [MakeMaker]         ; create Makefile.PL
 
       ; manifest (after all generated files)
       [Manifest]          ; create MANIFEST
 
       ; copy META.json back to repo dis
       [CopyFilesFromBuild]
       copy = META.json
       move = .gitignore
       copy = README.PATCHING
       copy = perlcritic.rc
 
       ; before release
       [Git::Check]        ; ensure all files checked in
       allow_dirty = dist.ini
       allow_dirty = Changes
       allow_dirty = README.pod
       allow_dirty = META.json
 
       [CheckPrereqsIndexed]    ; ensure prereqs are on CPAN
       [CheckChangesHasContent] ; ensure Changes has been updated
       [CheckExtraTests]   ; ensure xt/ tests pass
       [TestRelease]       ; ensure t/ tests pass
       [ConfirmRelease]    ; prompt before uploading
 
       ; releaser
       [UploadToCPAN]      ; uploads to CPAN
 
       ; after release
       [NextRelease]
 
       [Git::Commit] ; commit Changes (as released)
       allow_dirty = dist.ini
       allow_dirty = Changes
       allow_dirty = README.pod
       allow_dirty = META.json
 
       [Git::Tag]          ; tag repo with custom tag
       tag_format = release-%v
 
       [Git::Push]         ; push repo to remote
       push_to = origin

USAGE
    To use this PluginBundle, just add it to your dist.ini. You can provide
    the following options:

    *   "is_task" -- this indicates whether TaskWeaver or PodWeaver should
        be used. Default is 0.

    *   "auto_prereq" -- this indicates whether AutoPrereq should be used or
        not. Default is 1.

    *   "tag_format" -- given to "Git::Tag". Default is 'release-%v' to be
        more robust than just the version number when parsing versions

    *   "major_version" -- overrides the major version set by AutoVersion

    *   "fake_release" -- swaps FakeRelease for UploadToCPAN. Mostly useful
        for testing a dist.ini without risking a real release.

    *   "weaver_config" -- specifies a Pod::Weaver bundle. Defaults to
        @MITHALDU.

    *   "stopwords" -- add stopword for Test::PodSpelling (can be repeated)

    *   "no_critic" -- omit Test::Perl::Critic tests

    *   "no_spellcheck" -- omit Test::PodSpelling tests

    *   "gitignore" -- adds entries to be added to .gitignore (can be
        repeated)

    *   "exclude_match" -- regexes to exclude files from the dist (can be
        repeated)

    *   "prune_except" -- regexes to except files from being pruned as cruft
        (can be repeated)

SEE ALSO
    *   Dist::Zilla

    *   Dist::Zilla::Plugin::PodWeaver

    *   Dist::Zilla::Plugin::TaskWeaver

SUPPORT
  Bugs / Feature Requests
    Please report any bugs or feature requests through the issue tracker at
    <http://rt.cpan.org/Public/Dist/Display.html?Name=Dist-Zilla-PluginBundl
    e-MITHALDU>. You will be notified automatically of any progress on your
    issue.

  Source Code
    This is open source software. The code repository is available for
    public review and contribution under the terms of the license.

    <https://github.com/wchristian/dist-zilla-pluginbundle-mithaldu>

      git clone https://github.com/wchristian/dist-zilla-pluginbundle-mithaldu.git

AUTHORS
    *   David Golden <dagolden@cpan.org>

    *   Christian Walde <mithaldu@cpan.org>

COPYRIGHT AND LICENSE
    Christian Walde has dedicated the work to the Commons by waiving all of
    his or her rights to the work worldwide under copyright law and all
    related or neighboring legal rights he or she had in the work, to the
    extent allowable by law.

    Works under CC0 do not require attribution. When citing the work, you
    should not imply endorsement by the author.



Changes
Revision history for Dist-Zilla-PluginBundle-MITHALDU

1.141830  2014-07-02 23:40:35+02:00 Europe/Berlin
    - updates for dzil 5.019

1.120300  2012-01-30 14:36:47 Europe/Berlin

    - Dist::Zilla::App::cover is now a dependency to ensure it's available on
      the command line
    - .gitignore, README.PATCHING and perlcritic.rc are now generated during the
      build phase
    - added an option to exclude files by regex
    - release process now bumps the version in the release commit
    - AutoVersion instead of Git::NextVersion
    - minimize noise in development commits by restricting metadata updates to
      version and github urls to release time

0.023     2011-12-30 15:29:54 EST5EDT

    - change PortabilityTests and CompileTests to the newer Test::* naming
      scheme

0.022     2011-12-13 17:26:28 EST5EDT

    - re-enable Test::PodSpelling and added own 'stopwords' option as
      passthrough and 'no_spellcheck' to disable
    - Added Test::Perl::Critic and added 'no_critic' option to disable

0.021     2011-12-07 10:49:46 EST5EDT

    - make bugtracker pod section only list web link, not email; this
      allow easier modification for use with github instead of RT which
      does not provide an email option

0.020     2011-09-23 15:11:51 America/New_York

    - added code to copy README.pod and META.json to repo directory

0.019     2011-07-18 17:23:28 America/New_York

    - added Test::Version
    - removed unused prereqs

0.018     2011-07-11 21:52:26 America/New_York

    - swapped in InsertCopyright for Prepender
    - swapped in OurPkgVersion for PkgVersion
    - removed Repository to rely soly on GithubMeta

0.017     2011-02-06 21:57:53 EST5EDT

    - added Dist::Zilla::Plugin::CheckPrereqsIndexed

0.016     2011-01-31 22:14:12 EST5EDT

    - added Pod::Weaver::Section::Support and Bugtracker to the mix

0.015     2011-01-23 17:11:20 EST5EDT

    - updated prereqs, including missing Pod::Elemental::Transformer::List

0.014     2011-01-06 16:59:44 EST5EDT

    - bump Repository prereq to 0.17

0.013     2010-11-02 21:52:50 EST5EDT

    - bump MetaProvides::Package prereq and tell it to honor MetaNoIndex
    - add package 'DB' to MetaNoIndex

0.012     2010-10-29 16:42:37 EST5EDT

    - annotate example dist.ini in documents (suggeston from Tim Bunce)

0.011     2010-10-10 22:47:05 EST5EDT

    - replace BumpNextVersionFromGit with Git::NextVersion

0.010     2010-10-07 13:42:21 EST5EDT

    - adds GithubMeta to override Repository if repo is on github

    - replaces AutoPrereq (deprecated) with AutoPrereqs

0.009     2010-08-17 08:04:05 EST5EDT

    - set a minimum TaskWeaver prereq to work with Dist::Zilla 2.X

0.008     2010-08-11 17:50:46 EST5EDT

    - add Pod::Weaver::PluginBundle::DAGOLDEN (which is @RJBS plus wikidoc)

0.007     2010-05-26 12:49:30 America/New_York

    - bump Dist::Zilla prereq to 3.101450

    - bump Dist::Zilla::Plugin::Repository prereq to 0.13 for CPAN Meta
      spec version 2 compatibility

    - add MetaJSON plugin

0.006     2010-05-14 10:36:01 EST5EDT

    - fixes git commit/tag ordering; now commits dirty files
      after release, then tags, then updates Changes for next
      version and does a final commit before pushing

    - Pod spelling tests disabled pending fixes to stopwords and
      Pod::Weaver

0.005     2010-05-14 10:07:40 EST5EDT

    - fifth draft

0.004     2010-05-14 09:33:45 EST5EDT

    - fourth draft

0.003     2010-05-14 09:30:05 EST5EDT

    - third draft

0.002     2010-05-14 07:00:04 EST5EDT

    - second draft

0.001     2010-05-14 06:57:53 EST5EDT

    - first draft


lib/Pod/Weaver/PluginBundle/MITHALDU.pm
use strict;
use warnings;
package Pod::Weaver::PluginBundle::MITHALDU;
our $VERSION = '1.141830'; # VERSION

use Pod::Weaver::Config::Assembler;

# Dependencies
use Pod::Weaver::Plugin::WikiDoc ();
use Pod::Elemental::Transformer::List 0.101620 ();
use Pod::Weaver::Section::Support 1.001 ();

sub _exp { Pod::Weaver::Config::Assembler->expand_package($_[0]) }

my $repo_intro = <<'END';
This is open source software.  The code repository is available for
public review and contribution under the terms of the license.
END

my $bugtracker_content = <<'END';
Please report any bugs or feature requests through the issue tracker
at {WEB}.
You will be notified automatically of any progress on your issue.
END

sub mvp_bundle_config {
  my @plugins;
  push @plugins, (
    [ '@MITHALDU/WikiDoc',     _exp('-WikiDoc'), {} ],
    [ '@MITHALDU/CorePrep',    _exp('@CorePrep'), {} ],
    [ '@MITHALDU/Name',        _exp('Name'),      {} ],
    [ '@MITHALDU/Version',     _exp('Version'),   {} ],

    [ '@MITHALDU/Prelude',     _exp('Region'),  { region_name => 'prelude'     } ],
    [ '@MITHALDU/Synopsis',    _exp('Generic'), { header      => 'SYNOPSIS'    } ],
    [ '@MITHALDU/Description', _exp('Generic'), { header      => 'DESCRIPTION' } ],
    [ '@MITHALDU/Overview',    _exp('Generic'), { header      => 'OVERVIEW'    } ],

    [ '@MITHALDU/Stability',   _exp('Generic'), { header      => 'STABILITY'   } ],
  );

  for my $plugin (
    [ 'Attributes', _exp('Collect'), { command => 'attr'   } ],
    [ 'Methods',    _exp('Collect'), { command => 'method' } ],
    [ 'Functions',  _exp('Collect'), { command => 'func'   } ],
  ) {
    $plugin->[2]{header} = uc $plugin->[0];
    push @plugins, $plugin;
  }

  push @plugins, (
    [ '@MITHALDU/Leftovers', _exp('Leftovers'), {} ],
    [ '@MITHALDU/postlude',  _exp('Region'),    { region_name => 'postlude' } ],
    [ '@MITHALDU/Support',   _exp('Support'),
      {
        perldoc => 0,
        websites => 'none',
        bugs => 'metadata',
        bugs_content => $bugtracker_content,
        repository_link => 'both',
        repository_content => $repo_intro
      }
    ],
    [ '@MITHALDU/Authors',   _exp('Authors'),   {} ],
    [ '@MITHALDU/Legal',     _exp('Legal'),     {} ],
    [ '@MITHALDU/List',      _exp('-Transformer'), { 'transformer' => 'List' } ],
  );

  return @plugins;
}

# ABSTRACT: MITHALDU's default Pod::Weaver config
#
# This file is part of Dist-Zilla-PluginBundle-MITHALDU
#
#
# Christian Walde has dedicated the work to the Commons by waiving all of his
# or her rights to the work worldwide under copyright law and all related or
# neighboring legal rights he or she had in the work, to the extent allowable by
# law.
#
# Works under CC0 do not require attribution. When citing the work, you should
# not imply endorsement by the author.
#

1;

__END__

=pod

=head1 NAME

Pod::Weaver::PluginBundle::MITHALDU - MITHALDU's default Pod::Weaver config

=head1 VERSION

version 1.141830

=head1 DESCRIPTION

This is a L<Pod::Weaver> PluginBundle.  It is roughly equivalent to the
following weaver.ini:

   [-WikiDoc]
 
   [@Default]
 
   [Support]
   perldoc = 0
   websites = none
   bugs = metadata
   bugs_content = ... stuff (web only, email omitted) ...
   repository_link = both
   repository_content = ... stuff ...
 
   [-Transformer]
   transfomer = List

=for Pod::Coverage mvp_bundle_config

=head1 USAGE

This PluginBundle is used automatically with the CE<lt>@MITHALDUE<gt> L<Dist::Zilla>
plugin bundle.

=head1 SEE ALSO

=over

=item *

L<Pod::Weaver>

=item *

L<Pod::Weaver::Plugin::WikiDoc>

=item *

L<Pod::Elemental::Transformer::List>

=item *

L<Dist::Zilla::Plugin::PodWeaver>

=back

=head1 AUTHORS

=over 4

=item *

David Golden <dagolden@cpan.org>

=item *

Christian Walde <mithaldu@cpan.org>

=back

=head1 COPYRIGHT AND LICENSE


Christian Walde has dedicated the work to the Commons by waiving all of his
or her rights to the work worldwide under copyright law and all related or
neighboring legal rights he or she had in the work, to the extent allowable by
law.

Works under CC0 do not require attribution. When citing the work, you should
not imply endorsement by the author.

=cut


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.019.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Dist::Zilla configuration the way MITHALDU does it",
  "AUTHOR" => "David Golden <dagolden\@cpan.org>, Christian Walde <mithaldu\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-MITHALDU",
  "EXE_FILES" => [],
  "LICENSE" => "unrestricted",
  "NAME" => "Dist::Zilla::PluginBundle::MITHALDU",
  "PREREQ_PM" => {
    "CPAN::Meta" => 0,
    "Data::Section::Simple" => 0,
    "Dist::Zilla" => "4.3",
    "Dist::Zilla::App::Command::cover" => 0,
    "Dist::Zilla::Plugin::AutoVersion" => 0,
    "Dist::Zilla::Plugin::Bugtracker" => "1.102670",
    "Dist::Zilla::Plugin::CheckChangesHasContent" => 0,
    "Dist::Zilla::Plugin::CheckExtraTests" => 0,
    "Dist::Zilla::Plugin::CheckPrereqsIndexed" => "0.002",
    "Dist::Zilla::Plugin::CopyFilesFromBuild" => 0,
    "Dist::Zilla::Plugin::GithubMeta" => "0.10",
    "Dist::Zilla::Plugin::InsertCopyright" => "0.001",
    "Dist::Zilla::Plugin::MetaNoIndex" => 0,
    "Dist::Zilla::Plugin::MetaProvides::Package" => "1.11044404",
    "Dist::Zilla::Plugin::MinimumPerl" => 0,
    "Dist::Zilla::Plugin::OurPkgVersion" => "0.001008",
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::Plugin::ReadmeAnyFromPod" => "0.120051",
    "Dist::Zilla::Plugin::ReadmeFromPod" => 0,
    "Dist::Zilla::Plugin::StaticVersion" => 0,
    "Dist::Zilla::Plugin::TaskWeaver" => "0.101620",
    "Dist::Zilla::Plugin::Test::Compile" => 0,
    "Dist::Zilla::Plugin::Test::Perl::Critic" => 0,
    "Dist::Zilla::Plugin::Test::PodSpelling" => "2.001002",
    "Dist::Zilla::Plugin::Test::Portability" => 0,
    "Dist::Zilla::Plugin::Test::Version" => 0,
    "Dist::Zilla::PluginBundle::Filter" => 0,
    "Dist::Zilla::PluginBundle::Git" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "Dist::Zilla::Util::FileGenerator" => 0,
    "Moose" => "0.99",
    "Moose::Autobox" => 0,
    "Pod::Elemental::Transformer::List" => "0.101620",
    "Pod::Weaver::Config::Assembler" => 0,
    "Pod::Weaver::Plugin::WikiDoc" => 0,
    "Pod::Weaver::Section::Support" => "1.001",
    "Try::Tiny" => 0,
    "autodie" => "2.00",
    "namespace::autoclean" => "0.09",
    "strict" => 0,
    "strictures" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "File::Spec" => 0,
    "File::Temp" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::Exception" => "0.29",
    "Test::More" => "0.88"
  },
  "VERSION" => "1.141830",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "CPAN::Meta" => 0,
  "Data::Section::Simple" => 0,
  "Dist::Zilla" => "4.3",
  "Dist::Zilla::App::Command::cover" => 0,
  "Dist::Zilla::Plugin::AutoVersion" => 0,
  "Dist::Zilla::Plugin::Bugtracker" => "1.102670",
  "Dist::Zilla::Plugin::CheckChangesHasContent" => 0,
  "Dist::Zilla::Plugin::CheckExtraTests" => 0,
  "Dist::Zilla::Plugin::CheckPrereqsIndexed" => "0.002",
  "Dist::Zilla::Plugin::CopyFilesFromBuild" => 0,
  "Dist::Zilla::Plugin::GithubMeta" => "0.10",
  "Dist::Zilla::Plugin::InsertCopyright" => "0.001",
  "Dist::Zilla::Plugin::MetaNoIndex" => 0,
  "Dist::Zilla::Plugin::MetaProvides::Package" => "1.11044404",
  "Dist::Zilla::Plugin::MinimumPerl" => 0,
  "Dist::Zilla::Plugin::OurPkgVersion" => "0.001008",
  "Dist::Zilla::Plugin::PodWeaver" => 0,
  "Dist::Zilla::Plugin::ReadmeAnyFromPod" => "0.120051",
  "Dist::Zilla::Plugin::ReadmeFromPod" => 0,
  "Dist::Zilla::Plugin::StaticVersion" => 0,
  "Dist::Zilla::Plugin::TaskWeaver" => "0.101620",
  "Dist::Zilla::Plugin::Test::Compile" => 0,
  "Dist::Zilla::Plugin::Test::Perl::Critic" => 0,
  "Dist::Zilla::Plugin::Test::PodSpelling" => "2.001002",
  "Dist::Zilla::Plugin::Test::Portability" => 0,
  "Dist::Zilla::Plugin::Test::Version" => 0,
  "Dist::Zilla::PluginBundle::Filter" => 0,
  "Dist::Zilla::PluginBundle::Git" => 0,
  "Dist::Zilla::Role::PluginBundle::Easy" => 0,
  "Dist::Zilla::Util::FileGenerator" => 0,
  "File::Spec" => 0,
  "File::Temp" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Moose" => "0.99",
  "Moose::Autobox" => 0,
  "Pod::Elemental::Transformer::List" => "0.101620",
  "Pod::Weaver::Config::Assembler" => 0,
  "Pod::Weaver::Plugin::WikiDoc" => 0,
  "Pod::Weaver::Section::Support" => "1.001",
  "Test::Exception" => "0.29",
  "Test::More" => "0.88",
  "Try::Tiny" => 0,
  "autodie" => "2.00",
  "namespace::autoclean" => "0.09",
  "strict" => 0,
  "strictures" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



